modules/redis/options.go

```diff
@@ -0,0 +1,131 @@
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the 6380/tcp port to listen on for TLS connections and using a secure URL (rediss://).
+// It also creates a temporary directory to store the TLS certificates, which is removed
+// when the container is terminated.
+func WithTLS() Option {
+	return func(o *options) error {
+		o.tlsEnabled = true
+		return nil
+	}
+}
+
+// WithConfigFile sets the config file to be used for the redis container, and sets the command to run the redis server
+// using the passed config file
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	const defaultConfigFile = "/usr/local/redis.conf"
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: defaultConfigFile,
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+
+		// prepend the command to run the redis server with the config file, which must be the first argument of the redis server process
+		req.Cmd = append([]string{defaultConfigFile}, req.Cmd...)
+
+		return nil
+	}
+}
+
+// WithLogLevel sets the log level for the redis server process
+// See "[RedisModule_Log]" for more information.
+//
+// [RedisModule_Log]: https://redis.io/docs/reference/modules/modules-api-ref/#redismodule_log
+func WithLogLevel(level LogLevel) testcontainers.CustomizeRequestOption {
+	return testcontainers.WithCmdArgs("--loglevel", string(level))
+}
+
+// WithSnapshotting sets the snapshotting configuration for the redis server process. You can configure Redis to have it
+// save the dataset every N seconds if there are at least M changes in the dataset.
+// This method allows Redis to benefit from copy-on-write semantics.
+// See [Snapshotting] for more information.
+//
+// [Snapshotting]: https://redis.io/docs/management/persistence/#snapshotting
+func WithSnapshotting(seconds int, changedKeys int) testcontainers.CustomizeRequestOption {
+	if changedKeys < 1 {
+		changedKeys = 1
+	}
+	if seconds < 1 {
+		seconds = 1
+	}
+
+	return testcontainers.WithCmdArgs("--save", strconv.Itoa(seconds), strconv.Itoa(changedKeys))
+}
+
+// createTLSCerts creates a CA certificate, a client certificate and a Redis certificate,
+// storing them in the given temporary directory.
+func createTLSCerts() (*tlscert.Certificate, *tlscert.Certificate, *tlscert.Certificate, error) {
+	tmpDir, err := os.MkdirTemp(os.TempDir(), "testcontainers-redis-tls")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("mkdir temp dir: %w", err)
+	}
+	// ips is the extra list of IPs to include in the certificates.
+	// It's used to allow the client and Redis certificates to be used in the same host
+	// when the tests are run using a remote docker daemon.
+	ips := []net.IP{net.ParseIP("127.0.0.1")}
+
+	// Generate CA certificate
+	caCert := tlscert.SelfSignedFromRequest(tlscert.Request{
```

Comment:
bug: need to test for nil which indicates an error, more below.

We should also raise an issue to update SelfSignedFromRequest to return an explicit error, so we can provide that to the caller.

Reply:
I submitted a PR to add Err functions to tlscert: https://github.com/mdelapenya/tlscert/pull/1

When merged, I can release `v0.2.0` and directly use it here